src/server/pkg/storage/fileset/writer.go

```diff
@@ -198,15 +201,19 @@ func (w *Writer) Close() error {
 	}
 	// Write out the last index.
 	if w.lastIdx != nil {
+		idx := w.lastIdx
+		if !w.noUpload {
```

Comment:
Just for consistency, lets execute the idx callback before we write the index here. If you prefer the opposite, then switch the order in the callback.

Reply:
I have a weak preference for always after.  It seems like having an un-uploaded index in hand is less useful than an uploaded index.  I will change to match.